### PR TITLE
[Feature]: Snackbar error handling

### PR DESCRIPTION
**Pull request by OustanDing**
- `useSnackbar` hook for emitting messages via the snackbar
- Hook up snackbar with error catch blocks in frontend
- Add "Unauthorized" message to 401 errors

### CODE REVIEW

File: pages/performances/[id].js [hunk 1]
```diff
@@ -94,8 +96,9 @@ export default function PerformanceDetails({ session }) {
       setPerformance(formattedPerformance);
 
       await getAwards(response.data);
-    } catch {
+    } catch (err) {
       // Temporary solution, as error UI has not been implemented
```
Comment (feng-eric) [inline pages/performances/[id].js:100]: Can remove comment?

**Review by feng-eric — APPROVED** [review]
LGTM. Hopefully, we do not get any errors other than 400 or 401 from the backend (shouldn't be 🙃)

File: components/awards/ScoreBasedAwards.js [hunk 2]
```diff
@@ -19,11 +20,15 @@ import styles from '@styles/pages/Awards.module.scss'; // Page styles
...
 
+import useSnackbar from '@utils/useSnackbar'; // Snackbar
+
 const PAGE_SIZE = 20; // Rows per page
 
 // Page: Performances
 export default function Performances({ award }) {
+  const { snackbarError } = useSnackbar();
   const [event] = Event.useContainer();
+  const [session] = useSession();
```
Comment (mkanoria) [inline components/awards/ScoreBasedAwards.js:31]: I added session into this file in my last page, so could this be updated?

**Review by mkanoria — APPROVED** [review]
Just a small comment, looks great otherwise! Thanks for adding this!

File: pages/performances/[id].js (hunk 1 above)
Comment (OustanDing) [reply pages/performances/[id].js:100]: True lol thanks

File: components/awards/ScoreBasedAwards.js (hunk 2 above)
Comment (OustanDing) [reply components/awards/ScoreBasedAwards.js:31]: I just rebased tonight. Was it merged recently? Which commit was it?

Comment (mkanoria) [reply components/awards/ScoreBasedAwards.js:31]: https://github.com/uwblueprint/dancefest-web/commit/728deef9614aed2664cd4faa70ac4f4439fc2b1c

Comment (mkanoria) [reply components/awards/ScoreBasedAwards.js:31]: Looks like there's also a conflict because of that change

**Review by thepenguinco — APPROVED** [review]
LGTM.  After resolving the merge conflict of course.

File: components/awards/ScoreBasedAwards.js (hunk 2 above)
Comment (OustanDing) [reply components/awards/ScoreBasedAwards.js:31]: Ah okay, I'll resolve that. Thanks!

**Pull request merged**